projects/manager/src/app/datasets/dataset-editor/dataset-editor.component.ts

```diff
@@ -85,6 +97,60 @@ export class DatasetEditorComponent implements OnChanges {
         }
     }
 
+    removeTag(tag: string): void {
+        const tags: Array<string> = this.form.controls.tags.value;
+
+        const index = tags.indexOf(tag);
+        if (index > -1) {
+            tags.splice(index, 1);
+        }
+
+        this.form.markAsDirty();
+    }
+
+    addTag(event: MatChipInputEvent): void {
+        const tags: Array<string> = this.form.controls.tags.value;
+        const tag = event.value;
+        const input = event.input;
+
+        if (!isValidTag(tag)) {
+            return;
+        }
+
+        input.value = '';
+
+        tags.push(tag);
+
+        this.form.markAsDirty();
+    }
+
+    async saveProvenance(): Promise<void> {
+        if (!this.provenanceComponent.form.valid) {
+            return;
+        }
+
+        const provenance = this.provenanceComponent.getProvenance();
+
+        try {
+            await this.datasetsService.updateProvenance(this.datasetListing.name, provenance);
+            this.snackBar.open('Dataset provenance successfully updated.', 'Close', {duration: 2000});
+            this.provenanceComponent.form.markAsPristine();
+        } catch (error) {
+            const e = error as ResponseError;
+            const errorJson = await e.response.json().catch(() => ({}));
+            const errorMessage = errorJson.message ?? 'Updating dataset provenance failed.';
+            this.snackBar.open(errorMessage, 'Close', {panelClass: ['error-snackbar']});
```

Comment:
Can we create a method to be used for such errors?

I guess the way of extracting a geo engine error is always the same.

Reply:
7637c43867192241c6fd78ce44c33acc8159efe4